### PR TITLE
Fixed GHRepository.getFileContent() for path containing non-ASCII characters

### PR DESCRIPTION
**Pull request by reisners**
# Description 
using `URI.toASCIIString()` instead of `URI.toString()` in `GitHubRequest.urlPathEncode()`

# Notes:
- I tried but wasn't permitted to create a new branch in this repo. Therefore, I'm initiating this PR from a fork.
- test case and test data are included
- `mvn -Denable-ci clean install site` passes

### CODE REVIEW

**Comment by bitwiseman** [comment]
@reisners  You can manually shorten the paths: 
https://github.com/bitwiseman/github-api/commits/issue/803

File: src/test/resources/org/kohsuke/github/GHContentIntegrationTest/wiremock/testGetFileContentWithNonAsciiPath/__files/repos_github-api-test-org_ghcontentintegrationtest_contents_ghcontent-ro_a-dir-with-nonascii-entry_bornsteintxt-3.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "B??????rnstein.txt",
+  "path": "ghcontent-ro/a-dir-with-nonascii-entry/B??????rnstein.txt",
```
Comment (bitwiseman) [inline src/test/resources/org/kohsuke/github/GHContentIntegrationTest/wiremock/testGetFileContentWithNonAsciiPath/__files/repos_github-api-test-org_ghcontentintegrationtest_contents_ghcontent-ro_a-dir-with-nonascii-entry_bornsteintxt-3.json:3]: This is interesting.  What happens if you take this path string and call `repo.getFileContent()` on it?

Comment (reisners) [reply src/test/resources/org/kohsuke/github/GHContentIntegrationTest/wiremock/testGetFileContentWithNonAsciiPath/__files/repos_github-api-test-org_ghcontentintegrationtest_contents_ghcontent-ro_a-dir-with-nonascii-entry_bornsteintxt-3.json:3]: You mean against the real github? i haven't tried that but it surely won't work.

**Comment by reisners** [comment]
I've fixed the path, and all checks have now passed
(took me some time because in between the test data repo's org has changed from github-api-test-org to hub4j-test-org).
If you're ok with it, please merge it.

**Comment by bitwiseman** [comment]
Sorry about the org change, the timing was set by GitHub. Thanks for rolling with it.

File: src/test/resources/org/kohsuke/github/GHContentIntegrationTest/wiremock/testGetFileContentWithNonAsciiPath/__files/repos_github-api-test-org_ghcontentintegrationtest_contents_ghcontent-ro_a-dir-with-nonascii-entry_bornsteintxt-3.json (hunk 1 above)
Comment (bitwiseman) [reply src/test/resources/org/kohsuke/github/GHContentIntegrationTest/wiremock/testGetFileContentWithNonAsciiPath/__files/repos_github-api-test-org_ghcontentintegrationtest_contents_ghcontent-ro_a-dir-with-nonascii-entry_bornsteintxt-3.json]: It should work.  For other paths you can do exactly this.  It's a roundtrip thing.

File: src/test/java/org/kohsuke/github/GHContentIntegrationTest.java [hunk 2]
```diff
@@ -158,4 +160,11 @@ public void testMIMELonger() throws IOException {
                 + "123456789012345678901234567890123456789012345678901234567890");
         ghContentBuilder.commit();
     }
+
+    @Test
+    public void testGetFileContentWithNonAsciiPath() throws Exception {
+        final GHContent fileContent = gitHub.getRepository("hub4j-test-org/GHContentIntegrationTest")
+                .getFileContent("ghcontent-ro/a-file-with-\u00F6");
+        assertThat(IOUtils.readLines(fileContent.read(), StandardCharsets.UTF_8), hasItems("test"));
```
Comment (bitwiseman) [inline src/test/java/org/kohsuke/github/GHContentIntegrationTest.java:168]: I'm looking for a test something like this (I haven't run locally): 

```suggestion
        final GHRepository repo = gitHub.getRepository("hub4j-test-org/GHContentIntegrationTest");
        final GHContent fileContent = repo.getFileContent("ghcontent-ro/a-file-with-\u00F6");
        assertThat(IOUtils.readLines(fileContent.read(), StandardCharsets.UTF_8), hasItems("test"));
        
        final GHContent fileContent2 = repo.getFileContent(fileContent.getPath());
        assertThat(IOUtils.readLines(fileContent2.read(), StandardCharsets.UTF_8), hasItems("test"));
```

**Comment by reisners** [comment]
No worries, but please review the integration test. Since for lack of privileges I wasn't able to re-record some of the other test cases, I couldn't adapt the org globally. Instead, I create a separate GHRepository object just for the non-ascii test case. Ugly, but works.

File: src/test/java/org/kohsuke/github/GHContentIntegrationTest.java (hunk 2 above)
Comment (reisners) [reply src/test/java/org/kohsuke/github/GHContentIntegrationTest.java:168]: I just checked; your modification works on my machine. So yes, please go with it.

**Comment by bitwiseman** [comment]
Great job! Thanks!

**Pull request merged**